packages/shorebird_cli/test/src/commands/preview_command_test.dart

```diff
@@ -1267,6 +1278,376 @@ channel: ${DeploymentTrack.staging.channel}
+        when(() => androidReleaseArtifact.url)
+            .thenReturn(androidReleaseArtifactUrl);
+
+        when(
+          () => codePushClientWrapper.getReleaseArtifact(
+            appId: any(named: 'appId'),
+            releaseId: any(named: 'releaseId'),
+            arch: any(named: 'arch'),
+            platform: ReleasePlatform.android,
+          ),
+        ).thenAnswer((_) async => androidReleaseArtifact);
+
+        when(() => platform.isMacOS).thenReturn(true);
+      });
+
+      group('when the user choose ios', () {
+        setUp(() {
+          when(
+            () => logger.chooseOne<String>(
+              'Which platform would you like to preview?',
+              choices: any(named: 'choices'),
+            ),
+          ).thenReturn(ReleasePlatform.ios.displayName);
+
+          setupIOSShorebirdYaml()
+            ..createSync(recursive: true)
+            ..writeAsStringSync('app_id: $appId', flush: true);
+
+          when(
+            () => iosDeploy.installAndLaunchApp(
+              bundlePath: any(named: 'bundlePath'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).thenAnswer((_) async => ExitCode.success.code);
+        });
+
+        test('exits with success and calls install and launch app', () async {
+          final exitCode = await runWithOverrides(command.run);
+          expect(exitCode, equals(ExitCode.success.code));
+
+          verify(
+            () => iosDeploy.installAndLaunchApp(
+              bundlePath: any(named: 'bundlePath'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).called(1);
+        });
+
+        group('when the android release is not sideloadable', () {
+          setUp(() {
+            final unfilteredRelease = MockRelease();
+            when(() => unfilteredRelease.id).thenReturn(releaseId);
+            when(() => unfilteredRelease.version).thenReturn(releaseVersion);
+            when(() => unfilteredRelease.platformStatuses).thenReturn({
+              ReleasePlatform.ios: ReleaseStatus.active,
+              ReleasePlatform.android: ReleaseStatus.active,
+            });
+            when(() => release.platformStatuses).thenReturn({
+              ReleasePlatform.ios: ReleaseStatus.active,
+            });
+            when(
+              () => codePushClientWrapper.getReleases(
+                appId: any(named: 'appId'),
+                sideloadableOnly: true,
+              ),
+            ).thenAnswer((_) async => [release]);
+            when(
+              () => codePushClientWrapper.getReleases(
+                appId: any(named: 'appId'),
+              ),
+            ).thenAnswer((_) async => [unfilteredRelease]);
+          });
+
+          test('warns about the platform and goes directly to iOS', () async {
+            final exitCode = await runWithOverrides(command.run);
+            expect(exitCode, equals(ExitCode.success.code));
```

Comment:
I don't think we need to warn about Android if the user specified that they want to preview iOS